### ISSUE TITLE
How to handle removed entity field?

### ISSUE THREAD

**Issue by chengchen** [javers/javers#723]
Hello,

I am sorry if this question is already asked, and I did search before asking, the most relevant one I found is: https://github.com/javers/javers/issues/511

We are using javers to do audits in our company, and we are satisfied with it's simplicity. However, we had an issue recently regarding a simple removal of a field in an Hibernate entity. And this field itself is an entity as well. So to simplify a bit: Product entity had a field called Category, but Category has been removed completely from db as well as from the code. If I don't do any change in the javers tables, it will break simply with the exception:

```
org.javers.common.exception.JaversException: TYPE_NAME_NOT_FOUND: type name 'com.edgelab.marketdata.domain.Issuer' not found. If you are using @TypeName annotation, remember to register this class using JaversBuilder.withPackagesToScan(String) or JaversBuilder.scanTypeName(Class)
	at org.javers.core.metamodel.type.TypeMapperState.getClassByDuckType(TypeMapperState.java:70) ~[javers-core-3.11.5.jar:na]
	at org.javers.core.metamodel.type.TypeMapperState.getClassByTypeName(TypeMapperState.java:42) ~[javers-core-3.11.5.jar:na]
	at org.javers.core.metamodel.type.TypeMapper.getJaversManagedType(TypeMapper.java:168) ~[javers-core-3.11.5.jar:na]
	at org.javers.core.json.typeadapter.commit.GlobalIdTypeAdapter.parseEntity(GlobalIdTypeAdapter.java:117) ~[javers-core-3.11.5.jar:na]
	at org.javers.core.json.typeadapter.commit.GlobalIdTypeAdapter.parseInstanceId(GlobalIdTypeAdapter.java:59) ~[javers-core-3.11.5.jar:na]
	at org.javers.core.json.typeadapter.commit.GlobalIdTypeAdapter.fromJson(GlobalIdTypeAdapter.java:36) ~[javers-core-3.11.5.jar:na]
	at org.javers.core.json.typeadapter.commit.GlobalIdTypeAdapter.fromJson(GlobalIdTypeAdapter.java:15) ~[javers-core-3.11.5.jar:na]
	at org.javers.core.json.JsonConverterBuilder.lambda$registerJsonTypeAdapterForType$4(JsonConverterBuilder.java:168) ~[javers-core-3.11.5.jar:na]
....
```

I am wondering how should we proceed in this case without cleaning up massive amount of json snapshots? I also quickly browsed in the javers source code, I could not find a place to configure ignoring removed fields..

Could you help us with some hints? Or do you need some examples to reproduce?
Many thanks!

Chengchen

**Comment by bartoszwalacik** [comment]
hi, are you using the latest version?

**Comment by chengchen** [comment]
@bartoszwalacik we are using 3.11.5 but I will double check the latest one tomorrow.. Not sure if it will change anything though.

**Comment by bartoszwalacik** [comment]
looks like loading snapshots for removed classes is not supported. Think about contributing a PR to solve this problem. It shouldn't be hard. I can give you some hints

**Comment by chengchen** [comment]
@bartoszwalacik Sure I am willing to contribute. Please give some hints and I will do it when it's possible.

**Comment by bartoszwalacik** [comment]
@chengchen I have created the failing test case with issue reproduction
https://github.com/javers/javers/pull/725/files

**Comment by bartoszwalacik** [comment]
connected with 
https://github.com/javers/javers/issues/632

**Comment by chengchen** [comment]
@bartoszwalacik I tested my patch (workaround), at least it's working for our use cases. I am able to retrieve "removed" entity snapshots and even other entities contains "removed" entity.

**Comment by bartoszwalacik** [comment]
to be released in 3.11.7

**Comment by bartoszwalacik** [comment]
released in 3.11.7
